assets/js/googlesitekit/data/create-existing-tag-store.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Provides a datastore for getting existing tags
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_SITE } from '../datastore/site/constants';
+import { getExistingTagURLs, extractExistingTag } from '../../util/tag';
+
+const { createRegistryControl, createRegistrySelector } = Data;
+
+// Actions
+const FETCH_GET_EXISTING_TAG = 'FETCH_GET_EXISTING_TAG';
+const RECEIVE_GET_EXISTING_TAG = 'RECEIVE_GET_EXISTING_TAG';
+const WAIT_FOR_EXISTING_TAG = 'WAIT_FOR_EXISTING_TAG';
+
+/**
+ * Creates a store object that includes actions and selectors for getting existing tags.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {string}   type                 The data to access. One of 'core' or 'modules'.
+ * @param {string}   identifier           The data identifier, eg. a module slug like 'analytics'.
+ * @param {Object}   options              Options to consider for the store.
+ * @param {Array}    options.tagMatchers  The tag matchers used to extract tags from HTML.
+ * @param {Function} [options.isValidTag] Optional. Function to detect whether a tag is valid or not.
+ * @param {string}   [options.storeName]  Optional. Store name to use. Default is '{type}/{identifier}'.
+ * @return {Object} The existing tag store object, with additional `STORE_NAME` and
+ * INITIAL_STATE` properties.
+ */
+export const createExistingTagStore = ( type, identifier, {
+	tagMatchers,
+	isValidTag = ( tag ) => typeof tag === 'string' && tag.length > 0,
```

Comment:
Given the right `tagMatchers`, I think this check should always pass and this default could be sufficient. It's mostly to prevent `receiving` an invalid tag directly in tests. I wouldn't mind making it required but Felix originally mentioned it as an optional argument (https://github.com/google/site-kit-wp/pull/1721#issuecomment-657629547) so this is consistent with that. This is a private utility and we pass `isValidTag` functions for all of our uses in modules, so the default isn't really used outside of the tests, so I think it makes sense to remove.